### ISSUE TITLE
[deploy] Could not determine git HEAD null

### ISSUE THREAD

**Issue by maxandersen** [jreleaser/jreleaser#963]
if you do:

```
mkdir test
cd test
jreleaser init
git init 
jreleaser deploy
```

you get this error:

```
Could not determine git HEAD
null
```

which I assume is failing as no commits. since when done a git commit it tells me:

```
Could not determine remote
repository doesn't have a remote named 'origin'
```

**Comment by aalmiray** [comment]
That's correct. The `deploy` step requires running inside a git repository with at least 1 commit.

**Comment by maxandersen** [comment]
To be clear I get it's expected to not work. It's just that the error message is not as informative "null" as it is in other cases.

**Comment by aalmiray** [comment]
Aha. Did you get a stacktrace when the error occurred? Look into `out/jreleaser/trace.log`

**Comment by maxandersen** [comment]
```
java.io.IOException: repository doesn't have a remote named 'origin'
	at org.jreleaser.sdk.git.GitSdk.lambda$getRemote$1(GitSdk.java:92)
	at java.base/java.util.Optional.orElseThrow(Optional.java:408)
	at org.jreleaser.sdk.git.GitSdk.getRemote(GitSdk.java:92)
	at org.jreleaser.engine.context.ModelConfigurer.configure(ModelConfigurer.java:58)
	at org.jreleaser.engine.context.ContextCreator.create(ContextCreator.java:79)
	at org.jreleaser.cli.AbstractModelCommand.createContext(AbstractModelCommand.java:141)
	at org.jreleaser.cli.AbstractModelCommand.execute(AbstractModelCommand.java:82)
	at org.jreleaser.cli.AbstractCommand.call(AbstractCommand.java:36)
	at org.jreleaser.cli.AbstractModelCommand.call(AbstractModelCommand.java:48)
	at org.jreleaser.cli.AbstractCommand.call(AbstractCommand.java:29)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2314)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at org.jreleaser.cli.Main.execute(Main.java:72)
	at org.jreleaser.cli.Main.run(Main.java:57)
	at org.jreleaser.cli.Main.main(Main.java:48)
	at jreleaser_snapshot.main(jreleaser_snapshot.java:8)
```

**Comment by aalmiray** [comment]
Right now `deploy` has no dependencies to other steps (Maven deployers do not require previous cheksums, signing, changelog for example). However, other deployers might require checksums for example. Still, even if there were step dependencies the `deploy` step could be invoked without requiring a git repository. This is the case for the `assemble` step.

There are 2 issues here:
- decouple `deploy` from git.
- better error message when no git remote, git commit, or git repository is required.

@maxandersen thoughts?

**Comment by maxandersen** [comment]
Sorry. Didnt see the message before now. All good fixes!

**Comment by aalmiray** [comment]
Released in v1.4.0 -> https://github.com/jreleaser/jreleaser/releases/tag/v1.4.0
